10_rest/runnerdave/pkg/puppy/rest_test.go

```diff
@@ -0,0 +1,224 @@
+package puppy_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	puppy "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy"
+	store "github.com/anz-bank/go-course/10_rest/runnerdave/pkg/puppy/store"
+)
+
+var (
+	rs puppy.RestServer
+)
+
+func TestGetPuppy(t *testing.T) {
+	req, err := http.NewRequest("GET", "/api/puppy/1", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	rr := httptest.NewRecorder()
+	newPuppy := puppy.Puppy{
+		Breed:  "ULTRASURE",
+		Colour: "green",
+		Value:  2732.81,
+	}
+	db := store.NewSyncStore()
+	if err := db.CreatePuppy(newPuppy); err != nil {
+		panic(err)
```

Comment:
Similar, instead of `panic`, use `require.NoError`. Using `require` for test setup and `assert` for test verification makes it easy to see what is actually being tested, vs what is just needed to ensure the test runs in the right state.

Reply:
fixed